### PR TITLE
security: Add Content-Security-Policy

### LINKED ISSUE #611: Add a `Content-Security-Policy` header to prevent cross-site scripting (XSS) attacks
## Description

[Content Security Policy](https://scotthelme.co.uk/content-security-policy-an-introduction/) is an effective measure to protect our site from XSS attacks. By allowlisting (whitelisting) sources of approved content, we can prevent the browser from loading malicious assets.

## Implementation Model

In this process, first we will need to add an allowlist (whitelist) to every single embedded third party resources in our website, in order to be added to our CSP headers.

An example CSP header: https://github.com/leerob/leerob.io/blob/d3cb3af32b4be2a2fd2be5953d3c16d429f25943/next.config.js#L39-L49

## Tasks

- [x] Define which resources need to be added into the allowlist
- [x] Add the CSP header into `headers` key on `next.config.js` and `netlify.toml`

### PR DESCRIPTION
**Pull request by asaadam**
Closes #611 
## Description

Added a `Content-Security-Policy` header to prevent cross-site scripting (XSS).

## Current Tasks

 * [X]  Define which resources need to be added into the allowlist

 - defaultSrc : self
 - styleSrc : unsafe-inline
 - scriptSrc : self, .googletagmanager.com
 - imgSrc : self, https:, data: 

for `unsave-eval` on `scriptSrc`, we need to check if it's production or not. 
because in development webpack use unsave-eval.

* [X]  Add the CSP header into `headers` key on `next.config.js`
* [X]  Add the CSP header into `headers` key on `netlify.toml`

### CODE REVIEW

**Comment by sprabowo** [comment]
Screen Shot 2021-08-11 at 08 29 45

if we follow metrics from securityheaders it shows `A` result which is great https://securityheaders.com/?q=https%3A%2F%2Fdeploy-preview-634--wargabantuwarga.netlify.app%2F&followRedirects=on

better to add `frame-ancestor: 'none';` ref: https://observatory.mozilla.org/analyze/deploy-preview-634--wargabantuwarga.netlify.app @asaadam 
Screen Shot 2021-08-11 at 08 38 23

**Comment by asaadam** [comment]
> https://observatory.mozilla.org/analyze/deploy-preview-634--wargabantuwarga.netlify.app

done ya.

also i add `'unsafe-eval'` on `script-src` on `next.config.js`.
because it needs to be allowed while in development mode.

File: next.config.js [hunk 1]
```diff
@@ -3,7 +3,20 @@ const withBundleAnalyzer = require("@next/bundle-analyzer")({
...
 /** @type {import("next/dist/lib/config-shared").Header['headers']} */
+
+const ContentSecurityPolicy = `
+  default-src 'self';
+  script-src 'self' 'unsafe-inline' *.googletagmanager.com 'unsafe-eval';
+  style-src 'self' 'unsafe-inline';
+  img-src 'self' blob: data: https:;
+  frame-ancestors 'none';
+`;
+
```
Comment (resir014) [inline next.config.js:14]:
```suggestion
const ContentSecurityPolicy = `
  default-src 'self';
  script-src 'self' 'unsafe-inline' *.googletagmanager.com 'unsafe-eval';
  style-src 'self' 'unsafe-inline';
  img-src 'self' blob: data: https:;
  frame-ancestors 'none';
`;

/** @type {import("next/dist/lib/config-shared").Header['headers']} */
```

Don't split the JSDoc block from its variable. It is there for a reason.

**Review by resir014 — CHANGES_REQUESTED** [review]

File: next.config.js (hunk 1 above)
Comment (asaadam) [reply next.config.js]: sorry, and already fixed

**Review by resir014 — APPROVED** [review]

**Comment by zainfathoni** [comment]
@all-contributors please add @asaadam for security

**Pull request merged**